Move try except into run_and_combine_outputs function

jluethi opened the pull request:
The try except for the `subprocess.check_output` seems to be required to be in the function that calls `subprocess.check_output`. If it's in a function that calls a function that calls `subprocess.check_output` (as it is so far), it does not catch the output and therefore does not work.

This pull request would take it into the `run_and_combine_outputs` function. In my tests, this then provides the actual error messages to the console (see [this forum thread ](https://forum.image.sc/t/using-pyinstaller-on-a-program-that-uses-pyimagej/26642/5)for details).

The only thing I haven't been able to here is integrate it with the parser for the `parser.print_help(file=sys.stderr)`. Could that be done from within `run_and_combine_outputs`? Can the parser be initialized multiple times or passed to the `run_and_combine_outputs ` function?

ctrueden commented:
The Travis build failed due to a tabs-vs.-spaces issue. @jluethi Can you fix it, do `git commit -a --amend` and then `git push --force`? Or if this makes you uncomfortable, let me know and I'll take care of it.

ctrueden commented:
I purged the use of tabs, and force pushed it. Let's see if tests pass this time!

jluethi commented:
@ctrueden Sorry for the mess. I made quite a mess of the jgo.py on my end with lots of prints and debug statements and thought I'd be clever and just add the "small" change directly in the github web interface. That lead to all those tab & spaces issues.

Another issue I created in the copy pasting was forgetting the other except clauses in the main function. I cleaned it up and added another commit to move those too and now it passes the tests. 

But all the things that went to parser before have been commented out now

ctrueden commented:
Hmm, some more tabs got slipped in there, and the CI build didn't flag it. I fixed it with 4c6a0e23c62a10d338c882086a21eeae6efadece.

ctrueden commented:
No worries @jluethi—thanks a bunch for fixing the command invocation to be more verbose. I hope it helps debug your Windows issue.

jluethi commented:
@ctrueden Thanks a lot for the cleanup! :) And yeah, it helps trace the Windows issue. Though I still can't solve the issue at the moment because of some JDK vs. JRE issues or such (see imageJ forum).
Write once, run... ;)
